rpgpy/data.pyx

```diff
@@ -144,20 +146,26 @@ def _read_rpg_l0(file_name, header):
         fseek(ptr, n_dummy * 4, SEEK_CUR)
         fread(is_data, 1, n_levels, ptr)
 
+        # assign chirp number to ranges
+        chirp = np.digitize(range(n_levels), header['RngOffs'])
+
         for alt_ind in range(n_levels):
+            # loop over ranges
 
             if is_data[alt_ind] == 1:                
                 fseek(ptr, 4, SEEK_CUR)
+                # number of Doppler bins in this chirp
+                n_bins = header["SpecN"][chirp[alt_ind]-1]
 
                 if compression == 0:
 
                     n_points = n_samples_at_each_height[alt_ind]
-                    fread(&TotSpec[sample, alt_ind, 0], 4, n_points, ptr)
+                    fread(&TotSpec[sample, alt_ind, 0 + (n_spectra - n_bins)//2], 4, n_points, ptr)
```

Comment:
Calculate` (n_spectral - n_bins)//2` only once can assign it to a variable with some name that would somehow describe it. Also `0 +` is not needed here because it has no effect.